02_configure_host.sh

```diff
@@ -226,10 +226,12 @@ for IMAGE_VAR in $(env | grep "_LOCAL_IMAGE=" | grep -o "^[^=]*") ; do
     # Clone to ~ if not there already
     if [ -e "${REPOPATH}" ]; then
        cd "${REPOPATH}" || exit
+       git fetch upstream
+       git checkout "${BRANCH}"
```

Comment:
will not this fail if the where any uncommitted changes not sure if this case can happen

Reply:
Yes I'm not convinced about this either. Do we have `upstream` actually or is it just `origin`?

Reply:
these commands are executed on working directories that are created and by metal3-dev-env by default in the home directory just to use for the local image builds. This feature is only enabled if you do something like this ==> CAPM3_LOCAL_IMAGE=https://github.com/metal3-io/cluster-api-provider-metal3.git because the code checks the "://" section of the CAPM3_LOCAL_IMAGE and any other *_LOCAL_IMAGE variable. If it finds that substring then it enables this feature. So if you would need to specify a local directory where you can fiddle with the code then you just give the path to the code without the like CAPM3_LOCAL_IMAGE=/home/user/capm3

Reply:
+1, it is highly error-prone IMO

Reply:
This should be not touched by anyone manually, if a git address is used as image path then that means that you want to stick to given upstream branch. If you want to modify it locally then you just specify a repository path and there will be no git issues.

Reply:
btw @lentzi90  thans for spotting the "upstream" I have fixed it.